src/test/java/ProductDAOTest.java

```diff
@@ -0,0 +1,67 @@
+import model.dataaccess.*;
+import model.entities.Brand;
+import model.entities.Product;
+import model.entities.Section;
+import org.junit.*;
+
+
+import static main.MainClass.getConfig;
+import static org.junit.Assert.*;
+
+
+public class ProductDAOTest {
+
+
+    @Test
+    public void crudTest() {
+        getConfig();
+        IProductDAO pdao = new ProductDAO();
+        IBrandDAO bdao = new BrandDAO();
+        ISectionDAO sdao = new SectionDAO();
+
+        //----------------------ADD_TEST------------------------------
+
+        Section section = new Section("Electronics");
+        Brand brand = new Brand("LG", "telefoane");
+        bdao.add(brand);
+        sdao.add(section);
+        Brand brandTmp = bdao.getByName("LG");
+        Section sectionTmp = sdao.getByName("Electronics");
+        Product product = new Product("Telefon", "smartphone", 800, sectionTmp, brandTmp);
+        pdao.add(product);
+        boolean tr = false;
+        for (Product productTmp : pdao.getAll()) {
+            if (productTmp.getProductName().equals("Telefon") &&
+                    productTmp.getDetails().equals("smartphone") &&
+                    productTmp.getPrice() == 800 &&
+                    productTmp.getSection().getSectionName().equals("Electronics") &&
+                    productTmp.getBrand().getBrandName().equals("LG")) {
+                tr = true;
+            }
+        }
+        assertTrue(tr);
+
+        //---------------------UPDATE-TEST--------------------------
+        for (Product productTmp : pdao.getAll()) {

```

Comment:
nu are rost sa iterezi printr-un for ca sa gasesti produsul, il ai deja salvat pe sesiune, si instantiat in variabila product.

poti inlocui blockul de for cu

   product.setProductName("Smartphone");
        pdao.update(product);
        Product productReloadedFromBD = pdao.getById(product.getId());
        assertTrue(productReloadedFromBD.getProductName().equals("Smartphone"));